src/darwin/Framework/CHIP/ServerEndpoint/MTRServerEndpoint.mm

```diff
@@ -148,7 +171,21 @@ - (BOOL)addServerCluster:(MTRServerCluster *)serverCluster
     return YES;
 }
 
-- (BOOL)associateWithController:(MTRDeviceController *)controller
+#define MTR_DECLARE_LIST_ATTRIBUTE(attrID) \
+    DECLARE_DYNAMIC_ATTRIBUTE(attrID, ARRAY, 0, 0)
+
+static constexpr EmberAfAttributeMetadata sDescriptorAttributesMetadata[] = {
+    MTR_DECLARE_LIST_ATTRIBUTE(MTRAttributeIDTypeClusterDescriptorAttributeDeviceTypeListID),
```

Comment:
Same comment re the helper macro, especially since you're only using it for some attributes.

Reply:
Done.